internal/arduino/builder/internal/preprocessor/ctags.go

```diff
@@ -54,38 +57,38 @@ func PreprocessSketchWithCtags(sketch *sketch.Sketch, buildPath *paths.Path, inc
 
 	// Run GCC preprocessor
 	sourceFile := buildPath.Join("sketch", sketch.MainFile.Base()+".cpp")
-	gccStdout, gccStderr, err := GCC(sourceFile, ctagsTarget, includes, buildProperties)
-	verboseOutput.Write(gccStdout)
-	verboseOutput.Write(gccStderr)
-	normalOutput.Write(gccStderr)
+	result, err := GCC(sourceFile, ctagsTarget, includes, buildProperties)
+	verboseOutput.Write(result.Stdout())
+	verboseOutput.Write(result.Stderr())
+	normalOutput.Write(result.Stderr())
 	if err != nil {
 		if !onlyUpdateCompilationDatabase {
-			return normalOutput.Bytes(), verboseOutput.Bytes(), err
+			return Result{args: result.Args(), stdout: verboseOutput.Bytes(), stderr: normalOutput.Bytes()}, err
```

Comment:
I didn't understand this passage previously these functions returned `verboseOutput` and `normalOutput` where:
- `verboseOuput` had GCC stdout+stderr
- `normalOuput` had GCC stderr

now we have `stdout` and `stderr` where:
- `stdout` has GCC stdout+stderr
- `stderr` has GCC stderr

that works, but I don't like the improper assignment.

Maybe one solution could be to pass the verbose flag to the function, so:
1. we don't have to keep "verbose" and "non-verbose" output separated
2. we may return stdout and stderr unmixed